src/main/java/bio/terra/service/tabulardata/google/BigQueryPdao.java

```diff
@@ -1948,6 +2006,10 @@ public void queryForRowIds(
           QueryJobConfiguration.newBuilder(sqlQuery)
               .setDestinationTable(TableId.of(snapshotName, PDAO_TEMP_TABLE))
               .setWriteDisposition(JobInfo.WriteDisposition.WRITE_APPEND)
+              .setNamedParameters(
+                  Map.of(
+                      "transactionTerminatedAt",
+                      QueryParameterValue.timestamp(createdAt.toEpochMilli() * 1000)))
```

Comment:
I think this is
```suggestion
                      QueryParameterValue.timestamp(createdAt.truncatedTo(ChronoUnit.SECONDS))))
```

Reply:
And sadly there's no `toEpochNano()`

Reply:
It does? But milliseconds * 1000 is microseconds, isn't it?

Reply:
Err right.  Microseconds, sorry

Reply:
And confirmed in the docs https://cloud.google.com/bigquery/data-types

Reply:
Great, so `createdAt.truncatedTo(ChronoUnit.MICROS)` should work and let you get rid of the magic number.

Reply:
Ah this doesn't work because the API actually take a long

Reply:
I'll add a method to in here to make it more obvious